File: Iteration-2/src/StudentID.java
/**
 * This class is for creating unique IDs for students from departmnet code, registiration year and registiration order.
 */
public class StudentID {

    private Student student;
    private final String department = "1501"; // Only for computer engineering

    public StudentID(Student stu) {
        this.student = stu;
    }

    private String registrationYear() { // Finds reg year of students
        return String.valueOf((2021 - student.getYear()) % 100);
    }

    public String getStudentID() {
        return department + registrationYear() + String.format("%03d", student.getRegistrationOrder());
    }

    public String toString() {
        return getStudentID();
    }
}

File: Python Iteration/src/StudentID.py
class StudentID(object):
    def __init__(self, student):
        self.__student = student
        self.__department_code = "1501"

    def registration_year(self):
        reg_year = str((2021 - self.__student.year) % 100)
        return reg_year

    def get_student_id(self):
        stu_id = self.__department_code + self.registration_year() + "{:03}".format(self.__student.reg_order)
        return stu_id

    def __str__(self):
        return self.get_student_id()

